bip-p2qrh.mediawiki

```diff
@@ -50,26 +50,40 @@ The following table is non-exhaustive, but meant to be informative to the averag
+
+An increase in the witness discount must not be taken lightly, and must be resistant to applications that might take advantage of this discount such as storage of arbitrary data (commonly known as "inscriptions"), inflating chain state on nodes without a corresponding increase in economic activity. Such an increase would not only impact node runners, but those with inscriptions would also have the scarcity of their non-monetary assets affected.
+
+To address the risk of arbitrary data being stored using P2QRH (QuBit) addresses, very specific rules will be applied to spending from the witness stack in SegWit v3 outputs. A fixed signature size will be necessary for spending the output, and the output must be spendable to be considered valid within node consensus. A fixed signature size will also be helpful to disambiguate between signature types without an additional version byte, as SQIsign signatures are substantially smaller than FALCON signatures, and FALCON signatures are significantly larger than SQIsign signatures. Consequently, the correct signature algorithm can be inferred through byte length. The public key and signature will be pushed separately to the witness stack. Multiple signatures can be included in order to support multisig applications.
+
+Since only valid signatures can be committed to in a SegWit v3 witness, arbitrary data cannot be added by miners, as that would affect the consensus of their block. A CRQC operator is economically disincentivized from computing a spendable public key that matched arbitrary signature data due to the cost of that computation. That is because the cost of such a computation could prove quite substantial, rather than simply putting the arbitrary data within a Taproot witness. Doing the work to meet the requirement for it to be consensus-valid data would prove cost-prohibitive.
+
+Additionally, it should be noted, whether an output with a P2QRH spend script corresponds to a FALCON or SQIsign signature is not known until the output is spent.
+
+While it might be seen as a maintenance burden for bitcoin ecosystem devs to go from a single cryptosystem implementation to three distinct cryptosystems-- and it most certainly is-- the burden of a chain broken through extrinsic factors can be considered to override this constraint. Some things are just worth the extra effort, and having two PQC algorithms to fall back on rather than just one will be crucial given the experimental nature of SQIsign and the size burden of FALCON signatures. An increase in software maintenance everywhere signatures are used should be seen as an acceptable compromise for maintained integrity of bitcoin transfers within a regime of quantum advantage.
```

Comment:
```suggestion
While it might be seen as a maintenance burden for bitcoin ecosystem devs to go from a single cryptosystem implementation to three distinct cryptosystems—and it most certainly is—the burden of a chain broken through extrinsic factors can be considered to override this constraint. Some things are just worth the extra effort, and having two PQC algorithms to fall back on rather than just one will be crucial given the experimental nature of SQIsign and the size burden of FALCON signatures. An increase in software maintenance everywhere signatures are used should be seen as an acceptable compromise for maintained integrity of bitcoin transfers within a regime of quantum advantage.
```

Reply:
I'd prefer to use simpler formatting for sake of compatibility with regards to character sets.